distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -840,58 +841,119 @@ Text.drop_last count =
+        Pair True False -> Error.throw (Index_Out_Of_Bounds_Error range.end len)
+        Pair True True ->
+            if start>=end then (Range 0 0) else
+                iterator = BreakIterator.getCharacterInstance
+                iterator.setText text
 
-   ! What is a Character?
-     A character is defined as an Extended Grapheme Cluster, see Unicode
-     Standard Annex 29. This is the smallest unit that still has semantic
-     meaning in most text-processing applications.
+                loop index start_char end_char start_index =
+                    if index == end then (Range start_index (start_char)) else
+                        @Tail_Call loop (index + 1) end_char iterator.next (if index == start then start_char else start_index)
 
-   > Example
-     Make a new text from the first two characters of "boo".
+                loop 0 0 iterator.next Nothing
 
-         "boo".take_first 2
-Text.take_first : Integer -> Text
-Text.take_first count =
-    iterator = BreakIterator.getCharacterInstance
-    iterator.setText this
-    iterator.first
-    boundary = iterator.next count
-    if boundary == -1 then this else Text_Utils.substring this 0 boundary
+## ALIAS first, last, left, right, mid, substring
+   Creates a new Text by selecting the specified range of the input.
 
-## Creates a new text by selecting the last `count` characters of `this`,
-   returning `this` if `count` is greater than or equal to the length of `this`.
+   This can select a section of text from the beginning, end, or middle of the
+   input using various criteria defined by the range parameter.
 
    Arguments:
-   - count: The number of characters to take from the end of `this`.
+   - range: The section of the this text to return.
+     If a `Text_Sub_Range`, then the selection is interpreted following the rules of that type.
+     If a `Range`, the selection is specified by two indices, from and to.
+
+   Returns:
+   The part of the input as specified by the range parameter.
+
+   > Examples
+     Various different ways to take part of "Hello World!"
+
+         "Hello World!".take First.new == "H"
+         "Hello World!".take (First 5) == "Hello"
+         "Hello World!".take (First 0) == ""
+         "Hello World!".take Last.new == "!"
+         "Hello World!".take (Last 6) == "World!"
+         "Hello World!".take (Before " ") == "Hello"
+         "Hello World!".take (Before_Last "o") == "Hello W"
+         "Hello World!".take (After " ") == "World!"
+         "Hello World!".take (After_Last "o") == "rld!"
+         "Hello World!".take (While c->c!=" ") == "Hello"
+         "Hello World!".take (Range 3 5) == "lo"
+         "Hello World!".take (Range -3 -1) == "ld"
+         "Hello World!".take (Range -3 Nothing) == "ld!"
+         "Hello World!".take (Range 5 Nothing) == " World!"
+         "Hello World!".take (Range 5 12) == " World!"
+         "Hello World!".take (Range 12 12) == ""
```

Comment:
I'm not fully aware of the design there but I think the idea of examples is that (probably in the future) you can select one in the IDE and it will be placed on the stage - so then each example should probably be self contained instead of consisting of multiple variants.
That would make a LOT examples here, but I guess it can be useful to the users.
May worth asking @wdanilo or someone from IDE if the examples should be 'single separate units' or if grouping them like this is ok. Not sure how important that is since I'm not sure if/when this example insertion is going to be implemented, but I remember hearing something about it so would be good to clarify this.

Awesome set of examples by the way!

Reply:
will add to open questions